plugins/inputs/timex/timex.go

```diff
@@ -0,0 +1,82 @@
+//go:generate ../../../tools/readme_config_includer/generator
+//go:build linux
+
+package timex
+
+import (
+	_ "embed"
+	"fmt"
+	"time"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Timex struct {
+	Log telegraf.Logger `toml:"-"`
+}
+
+func (*Timex) SampleConfig() string {
+	return sampleConfig
+}
+
+func (*Timex) Gather(acc telegraf.Accumulator) error {
+	var timex unix.Timex
+	status, err := unix.Adjtimex(&timex)
+	if err != nil {
+		return fmt.Errorf("failed to get time adjtimex stats: %w", err)
+	}
+
+	// Check the return status for clock state
+	// https://github.com/torvalds/linux/blob/master/include/uapi/linux/timex.h
+	synced := status != unix.TIME_ERROR
```

Comment:
Shouldn't we add the returned status as a `status` tag? I would remove the `TIME_` of the return value description in the man page linked below and lowercase the text and add this as `status` tag.